integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/rest/test/header/ContentTypeTestCase.java

```diff
@@ -44,66 +47,72 @@
-		relativePath = relativePath.replaceAll("[\\\\/]", "/");
-		OMElement apiConfig = util.loadResource(relativePath);
-		addApi(apiConfig);
-	}
+    public static String contentType;
+    private static final int HTTP_STATUS_OK = 200;
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+
+        super.init();
+        String relativePath = "/artifacts/ESB/synapseconfig/esbjava2283/api.xml";
+        ESBTestCaseUtils util = new ESBTestCaseUtils();
+        relativePath = relativePath.replaceAll("[\\\\/]", "/");
+        OMElement apiConfig = util.loadResource(relativePath);
+        addApi(apiConfig);
+    }
 
     @Test(groups = {"wso2.esb"}, description = "Test different content types", dataProvider = "contentTypeProvider")
     public void testReturnContentType(String dataProviderContentType) throws Exception {
-	    contentType = dataProviderContentType;
-	    int port = 8089;
-	    server = HttpServer.create(new InetSocketAddress(port), 0);
-	    server.createContext("/gettest", new MyHandler());
-	    server.setExecutor(null); // creates a default executor
-	    server.start();
-	    DefaultHttpClient httpclient = new DefaultHttpClient();
-	    String url = "http://localhost:8480/serviceTest/test";
-	    HttpGet httpGet = new HttpGet(url);
-	    HttpResponse response = null;
-	    try {
-		    response = httpclient.execute(httpGet);
-	    } catch (IOException e) {
-		    log.error("Error Occurred while sending http get request. " + e);
-	    }
-	    log.info(response.getEntity().getContentType());
-	    log.info(response.getStatusLine().getStatusCode());
-
-	    assertEquals(response.getFirstHeader("Content-Type").getValue(), contentType, "Expected content type doesn't match");
-	    assertEquals(response.getStatusLine().getStatusCode(), 200, "response code doesn't match");
-
-	    server.stop(0);
+
+        contentType = dataProviderContentType;
+        int port = 8089;
+        HttpServer server = HttpServer.create(new InetSocketAddress(port), 0);
+        server.createContext("/gettest", new MyHandler());
+        server.setExecutor(null); // creates a default executor
+        server.start();
+        DefaultHttpClient httpclient = new DefaultHttpClient();
+        String url = "http://localhost:8480/serviceTest/test";
+        HttpGet httpGet = new HttpGet(url);
+        HttpResponse response = null;
+        try {
+            response = httpclient.execute(httpGet);
+        } catch (IOException e) {
+            log.error("Error Occurred while sending http get request. " , e);
+        }
+        log.info(response.getEntity().getContentType());
+        log.info(response.getStatusLine().getStatusCode());
+
+        assertEquals(response.getFirstHeader("Content-Type").getValue(), contentType, "Expected content type doesn't match");
+        assertEquals(response.getStatusLine().getStatusCode(), HTTP_STATUS_OK, "response code doesn't match");
+
+        server.stop(5);
```

Comment:
Shall we remove this magic number as well. Maybe a constant that suggest whether this a value in seconds or minutes...